HW4_smythe_full.py

```diff
@@ -0,0 +1,370 @@
+    else:
+        False
+    return col
+
+
+# predictor select for any dataset
+def predictor_select(ds):
+    predictors = ds.feature_names
+    # index_list = [i+1 for i in range(len(predictors))]
+    selection_df = pd.DataFrame(predictors, columns=["Predictor"])
+    selection_df.index += 1
+    print(selection_df)
+    pred_choice = input(
+        "Which predictors will you use? (use index numbers separate by a comma) :"
+    )
+    pred_choice = pred_choice.split(",")
+    for i in range(0, len(pred_choice)):
+        pred_choice[i] = int(pred_choice[i])
+    feature_list = [selection_df.at[i, "Predictor"] for i in pred_choice]
+
+    return feature_list
+
+
+def print_heading(title):
+    print("*" * 80)
+    print(title)
+    print("*" * 80)
+    return
+
+
+def list_to_df(from_df, to_df, col):
+    temp_list = from_df[col].to_list()
+    to_df[col] = pd.Series(temp_list, index=to_df.index)
+
+
+# Main functions
+def main():
+    start_t = datetime.now()
+    # set built-in dataset choices
+    ds1 = datasets.load_boston()
+    ds2 = datasets.load_iris()
+    ds3 = datasets.load_diabetes()
+    ds4 = datasets.load_digits()
+    ds5 = datasets.load_linnerud()
+    ds6 = datasets.load_wine()
+    ds7 = datasets.load_breast_cancer()
+
+    # Get user inputs -- add error handeling
+    # when response is continous then no 'target_names'
+    # when response is catagorical set 'target_names'
+    print(
+        """ Select SKLearn dataset to use: ~default is 3~
+                    1 - boston (response is continuous)
+                    2 - iris (response is catagorical [3 cats])
+                    3 - diabetes (response is continuous)
+                    4 - digits (response is an array)
+                    5 - linnerud (response is catagorical array of continuous)[3 cats])
+                    6 - wine (response is catagorical [3 cast])
+                    7 - breast cancer (response is catagorical {2cats/bool])
+                    8 - user choice
+                  """
+    )
+
+    choice = int(input("Which set? " or 3))
+
+    if choice == 1:
+        the_ds = ds1
+    elif choice == 2:
+        the_ds = ds2
+    elif choice == 3:
+        the_ds = ds3
+    elif choice == 4:
+        the_ds = ds4
+    elif choice == 5:
+        the_ds = ds5
+    elif choice == 6:
+        the_ds = ds6
+    elif choice == 7:
+        the_ds = ds7
+    elif choice == 8:
+        the_ds = input("direct path to dataset: ")
+    elif choice == "":
+        the_ds = ds3
+    else:
+        print(f"{choice} is an invalid entry. Please try again.")
+
+    print(f"You chose {choice}")
+    # turn source data into a pandas df
+    working_df = pd.DataFrame(the_ds.data, columns=the_ds.feature_names)
+    working_df["target"] = pd.Series(the_ds.target)  # chg
+    working_df.head()
+
+    # get column names
+
+    col_list = working_df.columns.values.tolist()
+    titles = [i.replace(")", "_").replace("(", "").replace(" ", "_") for i in col_list]

```

Comment:
This is unnecessary. You could just keep col_list = working_df.columns.values.tolist() and it should do the same thing.